layers/ip4.go

```diff
@@ -115,14 +115,16 @@ func (ip *IPv4) SerializeTo(b gopacket.SerializeBuffer, opts gopacket.SerializeO
 	binary.BigEndian.PutUint16(bytes[6:], ip.flagsfrags())
 	bytes[8] = ip.TTL
 	bytes[9] = byte(ip.Protocol)
-	if len(ip.SrcIP) != 4 {
-		return fmt.Errorf("invalid src IP %v", ip.SrcIP)
+	src := ip.SrcIP.To4()
```

Comment:
Why is checking for len == 4 not enough here?  If someone sets ipv4.SrcIP to an IPv6 which maps to a v4 address, that's still wrong.  Also, To4() may do extra memory allocs (not sure, haven't looked too closely).